Clarification for "Working with LINQ" tutorial

jmarkman opened the pull request:
## Summary

User "regal60740" (can't @ them for some reason) made some comments in the recently closed issue #7829 that I felt were pretty valid. I added a brief explanation on where the logging extension method goes and made a minor alignment change to some of the code using the LogQuery method to keep it consistent.

I'll also have to make a PR in dotnet/samples to fix the other issue they raised which highlighted some inconsistency between the markdown code snippet and the embedded one in the **Optimization** section

Edit: 
That pull request is available here: https://github.com/dotnet/samples/pull/469

Commit:
clarification
- explained where the logging extension method goes
- minor allignment with the LogQuery method

rpetrusha commented on docs/csharp/tutorials/working-with-linq.md:
```diff
@@ -257,8 +257,9 @@ Briefly, lazy evaluation states that the evaluation of a statement is not perfor
-Here's a log method that can be appended to any query to mark that the query executed.
+In your `Extensions.cs` file, type in or copy the method below. This new extension method will create a new file called `debug.log` within your project's directory and record what query is currently being executed to the log file. This extension method can be appended to any query to mark that the query executed.
 
+*`Extensions.cs`*
```
This can be removed. It doesn't conform to our standard presentation, which I think makes it confusing. The text also explains that this code goes in Extensions.cs.

jmarkman replied:
Sure. I think I added some more of these in my previous PR for this repo, so I'll make sure it's uniform.

rpetrusha commented on docs/csharp/tutorials/working-with-linq.md:
```diff
@@ -257,8 +257,9 @@ Briefly, lazy evaluation states that the evaluation of a statement is not perfor
 Remember that we generated the original deck using a LINQ query. Each shuffle is generated by performing three LINQ queries on the previous deck. All these are performed lazily. That also means they are performed again each time the sequence is requested. By the time you get to the 52nd iteration, you're regenerating the original deck many, many times. Let's write a log to demonstrate this behavi ...
 
-Here's a log method that can be appended to any query to mark that the query executed.
+In your `Extensions.cs` file, type in or copy the method below. This new extension method will create a new file called `debug.log` within your project's directory and record what query is currently being executed to the log file. This extension method can be appended to any query to mark that the query executed.
```
- "new extension method" --> "extension method" (to eliminate the duplication of "new")
- nit: will create --> creates
- nit: project's directory --> project directory
- record --> records

jmarkman replied:
All of this sounds good to me, I'll make these changes as well.

rpetrusha reviewed (changes requested):
Thanks, @jmarkman. Your expanded description of the example is extremely helpful. I've left some suggested changes.

Commit:
applied suggested changes from PR feedback
- removed the *`Extension.cs`* labels
- applied @rpetrusha's suggestions for the LogQuery paragraph

jmarkman commented:
It's quite alright, @rpetrusha. I'm happy to help in the first place.
